File: backend/models/analysis.py
from typing import List, Optional, Literal

from pydantic import BaseModel

from backend.models.evaluation import EvaluationScore

ApprovalStatus = Literal["pending", "approved", "disapproved"]
EvalRowStatus = Literal["completed", "pending", "failed", "not_evaluated"]


class AnalysisRow(BaseModel):
    filename: str
    path: str
    status: ApprovalStatus
    date: str
    created_at: float
    prompt: Optional[str] = None
    persona: Optional[str] = None
    eval_status: EvalRowStatus
    overall_score: Optional[float] = None
    scores: List[EvaluationScore] = []


class ApprovalBreakdown(BaseModel):
    approved: int
    disapproved: int
    pending: int
    approved_rate: float
    disapproved_rate: float
    pending_rate: float


class EvaluationBreakdown(BaseModel):
    evaluated: int
    not_evaluated: int
    failed: int
    evaluated_rate: float
    not_evaluated_rate: float


class AnalysisSummary(BaseModel):
    total: int
    approval: ApprovalBreakdown
    evaluation: EvaluationBreakdown
    avg_overall_score: Optional[float] = None


class AnalysisResponse(BaseModel):
    summary: AnalysisSummary
    items: List[AnalysisRow]
    total: int
    page: int
    pages: int
    per_page: int

File: frontend/src/types/analysis.ts
import type { EvaluationScore } from '@/types/evaluation'

export type ApprovalStatus = 'pending' | 'approved' | 'disapproved'
export type EvalRowStatus = 'completed' | 'pending' | 'failed' | 'not_evaluated'
export type AnalysisStatusFilter = 'all' | ApprovalStatus
export type EvaluatedFilter = 'all' | 'yes' | 'no'

export interface AnalysisRow {
  filename: string
  path: string
  status: ApprovalStatus
  date: string
  created_at: number
  prompt?: string | null
  persona?: string | null
  eval_status: EvalRowStatus
  overall_score?: number | null
  scores: EvaluationScore[]
}

export interface ApprovalBreakdown {
  approved: number
  disapproved: number
  pending: number
  approved_rate: number
  disapproved_rate: number
  pending_rate: number
}

export interface EvaluationBreakdown {
  evaluated: number
  not_evaluated: number
  failed: number
  evaluated_rate: number
  not_evaluated_rate: number
}

export interface AnalysisSummary {
  total: number
  approval: ApprovalBreakdown
  evaluation: EvaluationBreakdown
  avg_overall_score?: number | null
}

export interface AnalysisResponse {
  summary: AnalysisSummary
  items: AnalysisRow[]
  total: number
  page: number
  pages: number
  per_page: number
}
